Snabble/Core/Payment/Data/PayOneCreditCardData.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  PayOneData.swift
+//  Snabble
+//
+//  Created by Gereon Steffens on 27.09.21.
+//
+
+import Foundation
+
+// stores info from a PAYONE authorization
+struct PayoneCreditCardData: Codable, EncryptedPaymentData, Equatable, BrandedCreditCard {
+    // encrypted JSON string
+    let encryptedPaymentData: String
+    // serial # of the certificate used to encrypt
+    let serial: String
+
+    // name of this payment method for display in table
+    var displayName: String
+    let expirationDate: String // as "MM/YY"
+
+    let originType = AcceptedOriginType.payonePseudoCardPAN
+
+    let projectId: Identifier<Project>
+    let brand: CreditCardBrand
+
+    enum CodingKeys: String, CodingKey {
+        case encryptedPaymentData, serial, projectId, brand, displayName, expirationDate
+    }
+
+    private struct PayoneOrigin: PaymentRequestOrigin {
+        let pseudoCardPAN: String
+        let name: String
+        let userID: String
+    }
+
+    init?(gatewayCert: Data?, response: PayoneResponse, preAuthResult: PayonePreAuthResult, projectId: Identifier<Project>) {
+        let requestOrigin = PayoneOrigin(pseudoCardPAN: response.pseudoCardPAN, name: response.lastname, userID: preAuthResult.userID)
+
+        guard
+            let encrypter = PaymentDataEncrypter(gatewayCert),
+            let (cipherText, serial) = encrypter.encrypt(requestOrigin)
+        else {
+            return nil
+        }
+
+        self.brand = response.brand
+        self.projectId = projectId
+        self.encryptedPaymentData = cipherText
+        self.serial = serial
+        self.displayName = response.maskedCardPAN
+        self.expirationDate = response.cardExpireDate
+    }
+
+    var isExpired: Bool {
```

Comment:
Können wir das nicht mit der Logik aus TelecashCreditCard verbinden? Wozu haben wir zweimal identische Logik. Mein Vorschlag wäre es eventuell in dem neuen Protocol zu implementieren

Reply:
Die Logik ist leider nicht identisch, sondern nur ähnlich, daher fand ich es sicherer dafür den TC-Teil nicht anzufassen.

Reply:
Ich sehe hier keinen grundlegenden Unterschied. Für mich ist der Unterschied im Aufbereiten der Daten, da es einmal als String und bei der anderen Variante separat als Monate und Jahr gespeichert wird. 

Durch die Verbindung vom Daten- und ViewModel, passiert durch diese Kleinigkeit auch ein Unterschied in der Darstellung des ExpiredDate für den Nutzer. Bei Payone wird es als `MM/YY` angezeigt und für Telecash `MM/YYYY`. Ich vermute das ein Unterschied nicht gewollt ist.

Reply:
Der Unterschied in der Darstellung kommt daher dass uns TC und P1 das Datum unterschiedlich liefern, halt einmal als MMYY in einem String und einmal als MM und YYYY getrennt. Eine Anforderung die P1-Daten in MM/YYYY umzuwandeln ist mir nicht bekannt.